### PR TITLE
fix: clear conversation content can fail

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

if a conversation is not opening because of a crash in the mappers, then the user is not able to clear the content

### Solutions

refactor the use case and remove it's on reliant on mappers  

Needs releases with:

- [x] https://github.com/wireapp/kalium/pull/1849

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by typfel** [comment]
@MohamadJaara update the target branch to the MemberDAO refactoring branch so that we get usable diff of the changes.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt [hunk 1]
```diff
@@ -956,26 +955,29 @@ class UserSessionScope internal constructor(
...
-                isMessageSentInSelfConversation,
-            ),
-            DeleteForMeHandlerImpl(messageRepository, isMessageSentInSelfConversation),
-            DeleteMessageHandlerImpl(messageRepository, assetRepository, userId),
-            messageEncoder,
-            receiptMessageHandler,
-            userId
-        )
+        get()  {
+            val convRepo = conversationRepository
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt:959]: Delete, or what purpose this `val` serve?

File: persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/ConversationDAOTest.kt [hunk 2]
```diff
@@ -379,27 +381,57 @@ class ConversationDAOTest : BaseDatabaseTest() {
...
+                assertTrue(result.isEmpty())
+            }
 
-            assertTrue(result.isEmpty())
+            assertDAO.getAssetByKey("assetId").first().also { result ->
+                assertNull(result)
+            }
         }
 
-    // Mateusz : This test is failing because of some weird issue, I do not want to block this feature
```
Comment (typfel) [inline persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/ConversationDAOTest.kt:400]: Why remove this comment? The test is still commented out.

**Review by typfel — CHANGES_REQUESTED** [review]
Looking good, just some nit picks.

**Review by typfel — APPROVED** [review]

**Review by tmspzz — APPROVED** [review]

**Pull request merged**